app/models/helpers/piece_helper.rb

```diff
@@ -0,0 +1,79 @@
+module PieceHelper
+  def is_obstructed_horizontally?(row_destination, col_destination)
+    if self.row != row_destination
+      return false
+    end
+
+    # Pull in all pieces from same game and same row
+    pieces = self.game.pieces.where(row: self.row)
+
+    # Determine if pieces is moving right or left.
+    col_increment = self.column < col_destination ? 1 : -1
+    next_column = self.column + col_increment
+
+    # Loop through all columns between start and end position.
+    # Return true if any pieces on the row exist at specified column.
+    # (* col_increment) is done to allow < to always be correct operand.
+    while (next_column * col_increment) < (col_destination * col_increment)

```

Comment:
(as we discussed) This could potentially be implemented as:

```ruby
increasing = column < col_destination
start_col = column + (increasing ? 1 : -1)
end_col = col_destination + (increasing ? -1 : 1)
game.pieces.where(row: row, column: (start_col..end_col)).exists?
```

Or something like that.  Same idea for vertical as well.  This should run a single `COUNT` SQL query, which should be super fast.